search/pq.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+
+from sklearn.cluster import KMeans
+from sklearn.preprocessing import normalize
+
+from utils.distance import cos_similarity
+
+class ProductQuantisation:
+    """Product Quantisation class."""
+    n_subvectors = 8 # Num subvectors to split vector in for PQ
+    n_clusters = 256 # Num clusters for each subspace of subvectors
+
+    def __init__(self, dimension: int):
```

Comment:
subverters and cluster should be part of the constructor and default to the above